### ISSUE TITLE
Incorrect error message when `decl_macro` macro referes to `macro_rules!` macro

### ISSUE THREAD

**Issue by max-kamps** [rust-lang/rust#148414]
[Playground link](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2024&gist=e40e8ae9757738ebaf6f74fa9eee7953)

```rust
#![feature(decl_macro)]

macro_rules! inner {
    () => {}
}

macro outer() {
    inner!();
}

outer!();
```

This code results in this confusing and completely incorrect error message:
```rs
error: cannot find macro `inner` in this scope
  --> src/lib.rs:10:5
   |
10 |     inner!();
   |     ^^^^^ consider moving the definition of `inner` before this call
...
15 | outer!();
   | -------- in this macro invocation
   |
note: a macro with the same name exists, but it appears later
  --> src/lib.rs:5:14
   |
 5 | macro_rules! inner {
   |              ^^^^^
   = note: this error originates in the macro `outer` (in Nightly builds, run with -Z macro-backtrace for more info)
```
Obviously, the definition of `inner` doesn't "appear later" than the definition of `outer`. In fact, the problem has nothing to do with definition order. The real fix is to add `use inner;` anywhere after the definition of `inner`.

(As a side note, I don't know if this is the intended behavior of macros 2.0, or a bug / current limitation of the nightly feature)
